ssg/templates.py

```diff
@@ -285,25 +247,51 @@ def get_lang_for_rule(self, rule_id, rule_title, template, language):
         # definition ID so that the build system matches the generated
         # check with the rule.
         template_vars["_rule_id"] = rule_id
-        # checks and remediations are processed with a custom YAML dict
-        local_env_yaml = self.env_yaml.copy()
-        local_env_yaml["rule_id"] = rule_id
-        local_env_yaml["rule_title"] = rule_title
-        local_env_yaml["products"] = self.env_yaml["product"]
+        # Checks and remediations are processed with a custom YAML dict
+        local_env_yaml = {"rule_id": rule_id, "rule_title": rule_title}
+        if extra_env is not None:
+            local_env_yaml.update(extra_env)
+
+        return self.render_lang_file(template_name, template_vars, language, local_env_yaml)
+
+    def build_lang(self, rule_id, rule_title, template, lang, extra_env=None):
+        """
+        Builds templated content for a given rule for a given language.
+        Writes the output to the correct build directories.
+        """
+
+        filled_template = self.get_lang_contents(rule_id, rule_title, template, lang, extra_env)
+
+        ext = lang.file_extension
+        output_file_name = rule_id + ext
+        output_filepath = os.path.join(self.output_dirs[lang.name], output_file_name)
+
+        with open(output_filepath, "w") as f:
+            f.write(filled_template)
 
-        return self.build_lang_file(rule_id, template_name, template_vars,
-                                    language, local_env_yaml)
+    def build_rule(self, rule):
+        """
+        Builds templated content for a given rule for selected languages,
+        writing the output to the correct build directories.
+        """
+        extra_env = {}
+        if rule.identifiers is not None:
+            extra_env["cce_identifiers"] = rule.identifiers
+
+        langs_to_generate = self.get_langs_to_generate(rule.id_, rule.template)
+        for lang in langs_to_generate:
+            if lang not in self.templates[rule.template["name"]].langs or lang.name == "sce-bash":
+                continue
+            self.build_lang(rule.id_, rule.title, rule.template, lang, extra_env)
 
     def build_extra_ovals(self):
         declaration_path = os.path.join(self.templates_dir, "extra_ovals.yml")
         declaration = ssg.yaml.open_raw(declaration_path)
         for oval_def_id, template in declaration.items():
-            langs_to_generate = [languages["oval"]]
             # Since OVAL definition ID in shorthand format is always the same
             # as rule ID, we can use it instead of the rule ID even if no rule
             # with that ID exists
```

Comment:
This comment has become obsolete.

Reply:
@matejak No, it hasn't. We still pass `oval_def_id` instead of `rule_id` to the `self.build_lang` on line 294.